provider/pkg/resources/oidc_issuer_test.go

```diff
@@ -0,0 +1,423 @@
+// Copyright 2016-2025, Pulumi Corporation.
+
+package resources
+
+import (
+	"context"
+	"sort"
+	"testing"
+
+	"github.com/pulumi/pulumi-pulumiservice/provider/pkg/pulumiapi"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin"
+	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pulumi/pulumi-pulumiservice/provider/pkg/util"
+)
+
+// Regression test for issue #542 (https://github.com/pulumi/pulumi-pulumiservice/issues/542)
+// and issue #538 (https://github.com/pulumi/pulumi-pulumiservice/issues/538)
+//
+// This test verifies that the OIDC Issuer resource preserves the order of auth policies
+// returned by the API, and does not introduce spurious diffs through client-side reordering.
+//
+// The bug was caused by the sortPolicies() function which used gob.NewEncoder for comparison.
+// Gob encoding doesn't guarantee deterministic ordering of map keys, making the comparison
+// non-deterministic and causing false diffs when running `pulumi refresh` repeatedly.
+//
+// The fix (PR #542) removed the sortPolicies() function entirely, trusting the API to provide
+// consistent ordering. These tests ensure that policy ordering is preserved throughout the
+// Read() and Check() operations without any client-side reordering.
+
+// Mock implementation of pulumiapi.OidcClient for testing
+type OidcClientMock struct {
+	getOidcIssuerFunc   func(ctx context.Context, org string, id string) (*pulumiapi.OidcIssuerRegistrationResponse, error)
+	getAuthPoliciesFunc func(ctx context.Context, org string, id string) (*pulumiapi.AuthPolicy, error)
+}
+
+func (m *OidcClientMock) GetOidcIssuer(ctx context.Context, org string, id string) (*pulumiapi.OidcIssuerRegistrationResponse, error) {
+	if m.getOidcIssuerFunc != nil {
+		return m.getOidcIssuerFunc(ctx, org, id)
+	}
+	return nil, nil
+}
+
+func (m *OidcClientMock) GetAuthPolicies(ctx context.Context, org string, id string) (*pulumiapi.AuthPolicy, error) {
+	if m.getAuthPoliciesFunc != nil {
+		return m.getAuthPoliciesFunc(ctx, org, id)
+	}
+	return nil, nil
+}
+
+// Stub implementations for unused interface methods
+func (m *OidcClientMock) RegisterOidcIssuer(ctx context.Context, org string, req pulumiapi.OidcIssuerRegistrationRequest) (*pulumiapi.OidcIssuerRegistrationResponse, error) {
+	return nil, nil
+}
+
+func (m *OidcClientMock) UpdateOidcIssuer(ctx context.Context, org string, id string, req pulumiapi.OidcIssuerUpdateRequest) (*pulumiapi.OidcIssuerRegistrationResponse, error) {
+	return nil, nil
+}
+
+func (m *OidcClientMock) DeleteOidcIssuer(ctx context.Context, org string, id string) error {
+	return nil
+}
+
+func (m *OidcClientMock) UpdateAuthPolicies(ctx context.Context, org string, policyID string, req pulumiapi.AuthPolicyUpdateRequest) (*pulumiapi.AuthPolicy, error) {
+	return nil, nil
+}
```

Comment:
This will panic when unused functions are called, which IMO is clearer.

```suggestion
type OidcClientMock struct {
	pulumiapi.OidcClient
	getOidcIssuerFunc   func(ctx context.Context, org string, id string) (*pulumiapi.OidcIssuerRegistrationResponse, error)
	getAuthPoliciesFunc func(ctx context.Context, org string, id string) (*pulumiapi.AuthPolicy, error)
}

func (m *OidcClientMock) GetOidcIssuer(ctx context.Context, org string, id string) (*pulumiapi.OidcIssuerRegistrationResponse, error) {
	return m.getOidcIssuerFunc(ctx, org, id)
}

func (m *OidcClientMock) GetAuthPolicies(ctx context.Context, org string, id string) (*pulumiapi.AuthPolicy, error) {
	return m.getAuthPoliciesFunc(ctx, org, id)
}
```